test/test_all_fields.py

```diff
@@ -1,25 +1,27 @@
-import tap_tester.connections as connections
-import tap_tester.runner as runner
-import tap_tester.menagerie as menagerie
+import base
+
 from base import ZendeskTest
+from tap_tester import connections, menagerie, runner
+
 
 class ZendeskAllFields(ZendeskTest):
     """Ensure running the tap with all streams and fields selected results in the replication of all fields."""
-     
+
     def name(self):
         return "zendesk_all_fields"
 
+    @base.tt_base.skipUnless(base.JIRA_CLIENT.get_jira_issue_status("TDL-20862") == "Done", "TDL-20862")
```

Comment:
Since this is code that's meant to de deleted once a jira ticket is done the code we add here isn't super important.

---

While you did remove all of the duplication that I mentioned in my first review, I think there's still room for improvement here.

1. `tap_tester.base.skipUnless` has a very weird API right now: given a boolean and some jira ticket string, skip a test or throw an error. And the boolean is always the same (is some status equal to `"Done"`) and the jira ticket string is always duplicated.

2. Using an import that another file imported sounds like more trouble than it's worth.

So my suggestion is this for every test file:

```suggestion
    @base.skipUntilDone("TDL-20862")
```

Add one function to zendesk `base` that knows how to properly call `tap_tester.skipUnless`
- I'm not dead set on this name, but I think it captures the intent of the function well
- There's no use of `base.tt_base` in any test now

The implementation of `base.skipUntilDone` will probably look something like this
```python
JIRA_CLIENT = jira_client({ **jira_config })

class ZendeskBase:

    def skipUntilDone(jira_ticket):

        def wrap(test_method):
            is_done = JIRA_CLIENT.get_jira_issue_status(jira_ticket) == "Done"
            return tap_tester.base.skipUnless(is_done, jira_ticket)(test_method)

        return wrap
```

Reply:
But again, all of this goes away some day, and since things are passing, I'm happy to just approve. I just think there's a lot to learn from this PR as maintainers and consumers of tap-tester

Reply:
Could not call the new function directly from base without providing the parent class.  Also tap_tester is not imported as a stand alone so inside the function I replaced that with the existing import of tt_base.  Let me know if this next commit works or we need another round.